### PR TITLE
feat(core): Map/Unmap SGs and Load Balancers

### PR DESCRIPTION
**Pull request by stuart-pollock**
This change contains preparation for adding pipeline stages to Deck for mapping load balancers to  server groups, and also for unmapping load balancers from server groups.

### CODE REVIEW

File: orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/instance/DeregisterInstancesFromLoadBalancerTaskSpec.groovy [hunk 1]
```diff
@@ -43,44 +45,55 @@ class DeregisterInstancesFromLoadBalancerTaskSpec extends Specification {
 
   def "creates a deregister instances from load balancer task based on job parameters"() {
```
Comment (claymccoy) [inline orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/instance/DeregisterInstancesFromLoadBalancerTaskSpec.groovy]: these test names would read slightly better the task name was set apart, maybe by single quotes

File: orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/instance/DeregisterInstancesFromLoadBalancerTaskSpec.groovy [hunk 2]
```diff
@@ -43,44 +45,55 @@ class DeregisterInstancesFromLoadBalancerTaskSpec extends Specification {
...
     given:
-    def operations
-    task.trafficGuard = Mock(TrafficGuard)
-    task.kato = Mock(KatoService) {
-      1 * requestOperations("abc", _) >> {
-        operations = it[1]
-        rx.Observable.from(taskId)
+      def operations
+      task.trafficGuard = Mock(TrafficGuard)
```
Comment (claymccoy) [inline orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/instance/DeregisterInstancesFromLoadBalancerTaskSpec.groovy]: This test is very similar to the AbstractLoadBalancerRegistrationTaskSpec aside from the task being used and the traffic guard. I'm wondering if some of the data and asserts should be reused.

**Comment by anotherchrisberry** [comment]
Since these are specific to Cloud Foundry, it's probably better to put these into a provider-specific package: `com.netflix.spinnaker.orca.clouddriver.pipeline.providers.cloudfoundry`.

There are already packages for `appengine`, `aws`, `gce`, and `titus`.

I agree with Clay that it's unfortunate we need full-blown stages/tasks for these boilerplate passthroughs to Clouddriver, but that is the world we live in.

**Comment by ajordens** [comment]
`I agree with Clay that it's unfortunate we need full-blown stages/tasks for these boilerplate passthroughs to Clouddriver, but that is the world we live in.`

We just need a contribution that targets this ... it's not an impossible to do. It's been on my back burner for awhile but still something I'd like to see.

**Pull request merged**